lib/_http_client.js

```diff
@@ -216,77 +216,69 @@ function ClientRequest(options, cb) {
   this.maxHeadersCount = null;
 
   var called = false;
-  if (self.socketPath) {
-    self._last = true;
-    self.shouldKeepAlive = false;
+
+  const oncreate = (err, socket) => {
```

Comment:
The function was using a closure around `self`. I modified it to use lexical `this` with the arrow function.